java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightClient.java

```diff
@@ -353,10 +355,11 @@ public void onCompleted() {
    */
   public ExchangeReaderWriter doExchange(FlightDescriptor descriptor, CallOption... options) {
     Preconditions.checkNotNull(descriptor, "descriptor must not be null");
-    final io.grpc.CallOptions callOptions = CallOptions.wrapStub(asyncStub, options).getCallOptions();
+    FlightServiceStub stub = CallOptions.wrapStub(asyncStub, options);
+    final io.grpc.CallOptions callOptions = stub.getCallOptions();
 
     try {
-      final ClientCall<ArrowMessage, ArrowMessage> call = interceptedChannel.newCall(doExchangeDescriptor, callOptions);
+      final ClientCall<ArrowMessage, ArrowMessage> call = stub.getChannel().newCall(doExchangeDescriptor, callOptions);
```

Comment:
Is it possible to factor out a helper that can take care of the wrapStub - getChannel - newCall dance?

Reply:
it'd be nice to add something like:

```
    public <RequestT, ResponseT> ClientCall<RequestT, ResponseT> newCall(MethodDescriptor<RequestT, ResponseT> descriptor) {
      return getChannel().newCall(descriptor, getCallOptions());
    }
```

to the stub so we could just do `stub.newCall(descriptor)`, but it looks like that class is auto-generated so I'm not sure where to add it.

Reply:
It can just be a private method, no?

Reply:
Oh I meant this would be great as a public method in `FlightServiceStub`. There's a lot of code in there that could use it, but I believe that is an auto-generated source file so I don't know how to change it.

I went ahead and created a private method in `FlightClient` to do the same.

Reply:
It's generated by gRPC, so it's not under our control